common/src/main/java/io/netty/util/internal/ObjectUtil.java

```diff
@@ -108,7 +195,38 @@ public static long checkInRange(long l, long start, long end, String name) {
      */
     public static <T> T[] checkNonEmpty(T[] array, String name) {
         checkNotNull(array, name);
-        checkPositive(array.length, name + ".length");
+        //No String concatenation for check
+        if (array.length == 0) {
+            throw new IllegalArgumentException(String.format(EX_PARAM_MUST_NOT_BE_EMPTY, name));
+        }
+        return array;
+    }
+
+    /**
+     * Checks that the given argument is neither null nor empty.
+     * If it is, throws {@link NullPointerException} or {@link IllegalArgumentException}.
+     * Otherwise, returns the argument.
+     */
+    public static byte[] checkNonEmpty(byte[] array, String name) {
+        checkNotNull(array, name);
+        //No String concatenation for check
+        if (array.length == 0) {
```

Comment:
could merge with above check

Reply:
What do you mean with "above":
Do you want to merge checkNotNull(1) and array.length(2) because 2 will throw a NullPointer? This is not the idea - the NullPointerEx would be without "name" in it's message.
Or do you want to have a type conversion in any check above? <T> T[] checkNonEmpty(T[] array will not help - the compiler refuses to use this.

Reply:
You mean:
`if (checkNotNull(array, name).length == 0) {`
?

Reply:
yes